Fix the case when algo is an object

kjhman21 opened the pull request:
When we call Crypto.subtle.verify, we need to pass verifyAlgorithm like this:

```javascript
    const verifyAlgorithm = {
        name: 'ECDSA',
        hash: { name: 'SHA-256' },
    };
    return crypto.subtle.verify(verifyAlgorithm, key, signature, data);
```

In this case, `algo.toString().toLowercase()` will not work.

This PR resolves the problem.

boorad commented:
hey, thanks for this...  can you paste a screenshot of the example app test results in this PR?

kjhman21 commented:
@boorad Hi, sorry but it is unclear what I need to paste. Are you talking about error screen of this?

Here is what I have encountered this problem.

1. I installed `react-native-quick-crypto` in the main App() like below
```js
import { install } from 'react-native-quick-crypto';

export default function App() {
...
  install()
...
}
```

2. There are some usages in the library that I imported like this.
```js
    const verifyAlgorithm = {
        name: 'ECDSA',
        hash: { name: subtleAlg },
    };
    return WebCrypto.subtle.verify(verifyAlgorithm, key, signature, data);
```

3. In my investigation, I found that it eventually calls signVerify in `react-native-quick-crypto/lib/module/subtle.js`.
<img width="710" alt="image" src="https://github.com/user-attachments/assets/a510830e-a83c-42d3-833d-73f496d3d85b">


4. In this function, it calls ecdsaSignVerify in `react-native-quick-crypto/lib/module/ec.ts`
<img width="887" alt="image" src="https://github.com/user-attachments/assets/8565c816-fca5-4159-957e-90f7fc3002ee">


5. In this function, it calls normalizeHashName in `react-native-quick-crypto/lib/modules/Hashnames.js`
<img width="637" alt="image" src="https://github.com/user-attachments/assets/f5395b8a-43a2-4c48-9a3c-68bc53d456c9">

6. I added print code via console.log, and here is the result.
<img width="310" alt="image" src="https://github.com/user-attachments/assets/bb50d9c0-5afe-47cc-8aa1-ced68e6eae0f">


After I added the code in this PR, here is the result.
<img width="335" alt="image" src="https://github.com/user-attachments/assets/8ab65c1e-e462-4e47-b1fc-04f89e4793cd">


Hope this helps you understand the change. Thanks.

boorad commented:
I just need a screenshot of the example app with all of the tests run.  It shows me your change didn't break anything else.  See #443

kjhman21 commented:
@boorad Okay. got the point. I will prepare it soon!

kjhman21 commented:
**BEFORE**
<img width="449" alt="image" src="https://github.com/user-attachments/assets/d4028ca6-fb2b-43a9-a602-9b4bde2f0734">

**AFTER**
<img width="450" alt="image" src="https://github.com/user-attachments/assets/4f730998-87e7-4ebf-a4e3-ad338e47ec71">

kjhman21 commented:
seems like changing base branch is not working. I will create another PR.

kjhman21 commented:
@boorad Please take a look of the PR #457 instead.
